### PR TITLE
Begin migration to test plans, split unit and integration tests

### PR DESCRIPTION
**Pull request by pjleonard37**
Migrate tests to test plans. 

## Pull request checklist:
 - [x] Describe the changes in this PR, especially public API changes.
 - [ ] Include before/after visuals or gifs if this PR includes visual changes.
    
 - [ ] Write tests for all new functionality. If tests were not written, please explain why.
 - [ ] Add documentation comments for any added or updated public APIs.
 - [ ] Add any new public, top-level symbols to the Jazzy config's `custom_categories` (scripts/doc-generation/.jazzy.yaml)
 - [ ] Add a changelog entry to to bottom of the relevant section (typically the `## main` heading near the top).
 - [ ] Update the guides (internal access only), README.md, and DEVELOPING.md if their contents are impacted by these changes.
 - [ ] If this PR is a `v10.[version]` release branch fix / enhancement, merge it to `main` first and then port to `v10.[version]` release branch.

PRs must be submitted under the terms of our Contributor License Agreement [CLA](https://github.com/mapbox/mapbox-maps-ios/blob/main/CONTRIBUTING.md#contributor-license-agreement).

### CODE REVIEW

File: Apps/DebugApp/DebugApp.xcodeproj/xcshareddata/xcschemes/DebugApp.xcscheme [hunk 1]
```diff
@@ -41,8 +41,12 @@
             isEnabled = "YES">
          </AdditionalOption>
       </AdditionalOptions>
-      <Testables>
-      </Testables>
+      <TestPlans>
+         <TestPlanReference
+            reference = "container:../../TestPlans/DebugApp.xctestplan"
```
Comment (OdNairy) [inline Apps/DebugApp/DebugApp.xcodeproj/xcshareddata/xcschemes/DebugApp.xcscheme:46]: DebupApp testplan is empty. Lets replace it with UnitTests plan

Comment (pjleonard37) [reply Apps/DebugApp/DebugApp.xcodeproj/xcshareddata/xcschemes/DebugApp.xcscheme:46]: 👍  -- updated in most recent commit

File: Apps/Apps.xcworkspace/contents.xcworkspacedata [hunk 2]
```diff
@@ -13,4 +13,22 @@
...
+   </FileRef>
+   <FileRef
+      location = "group:../TestPlans/DebugApp.xctestplan">
+   </FileRef>
+   <FileRef
+      location = "group:../TestPlans/UnitTests.xctestplan">
+   </FileRef>
+   <FileRef
+      location = "group:../TestPlans/IntegrationTests.xctestplan">
+   </FileRef>
```
Comment (OdNairy) [inline Apps/Apps.xcworkspace/contents.xcworkspacedata:33]: We can avoid explicit references as the will be available through folder in SPM tree

**Review by OdNairy — APPROVED** [review]
LGTM

**Pull request merged**